osdep/terminal-unix.c

```diff
@@ -427,6 +427,9 @@ static void *terminal_thread(void *ptr)
                 process_input(input_ctx, false);
             }
         }
+        if (fds[1].revents & POLLHUP) {
```

Comment:
mpv doesn't use braces for single line `if`s.

Reply:
Also, I'd move this check after the other `if (r == 0) process_input(...)` line. Just in case there was still some input left in the buffer to process, since AFAIK `POLLIN` and `POLLHUP` can be true in parallel. E.g when there's still data left in the pipe, but the fd has been closed from the other end.